be/src/vec/functions/array/function_array_binary.h

```diff
@@ -39,7 +39,11 @@ class FunctionArrayBinary : public IFunction {
     DataTypePtr get_return_type_impl(const DataTypes& arguments) const override {
         DCHECK(is_array(arguments[0])) << arguments[0]->get_name();
         DCHECK(is_array(arguments[1])) << arguments[1]->get_name();
-        DCHECK(arguments[0]->equals(*arguments[1]))
+        auto left_nested_type = remove_nullable(
```

Comment:
No need to remove_nullable if use_default_implementation_for_nulls() return true.

Reply:
a p0 regression test case "select arrays_overlap([1,2,3,null], [3,4,5])" and "select array_intersect([1,2,3], [2,3,4,null])" would trigger the DCHECK because the data type is array<nullable<int>> and array<int>